proxy/router/src/lib.rs

```diff
@@ -51,6 +52,13 @@ pub trait Recognize {
 
 pub struct Single<S>(Option<S>);
 
+/// Whether or not the service to a given key may be cached.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum Cachability<T> {
```

Comment:
I don't have a better suggestion for a name right now, but my first instinct is that we should reserve names like `Cachability` for things related to HTTP-level caching (Cache-Control, etc.).

Reply:
That's fair. The phrase "cache" is used repeatedly in `router` & I thought it seemed clearer than the name I gave this originally, `Reusability`.